WooCommerce/Classes/ViewRelated/Products/Add Product/ManualProductTypeOptions.swift

```diff
@@ -6,37 +6,70 @@ struct ManualProductTypeOptions: View {
     private let productTypes: [BottomSheetProductType]
     private let onOptionSelected: (BottomSheetProductType) -> Void
 
+    /// Grouped product types based on their product category
+    private let groupedProductTypes: [(BottomSheetProductType.ProductCreationCategory, [BottomSheetProductType])]
+
     init(productTypes: [BottomSheetProductType],
          onOptionSelected: @escaping (BottomSheetProductType) -> Void) {
         self.productTypes = productTypes
         self.onOptionSelected = onOptionSelected
+
+        self.groupedProductTypes = Constants.productCategoriesOrder.compactMap { category in
+            let currentCategoryProductTypes = productTypes.filter { $0.productCreationCategory == category }
+            return currentCategoryProductTypes.isEmpty ? nil : (category, currentCategoryProductTypes)
+        }
     }
 
     var body: some View {
-        ForEach(productTypes) { productType in
-            HStack(alignment: .top, spacing: Constants.margin) {
-                Image(uiImage: productType.actionSheetImage.withRenderingMode(.alwaysTemplate))
-                    .font(.title3)
-                    .foregroundStyle(.secondary)
-
-                VStack(alignment: .leading, spacing: Constants.verticalSpacing) {
-                    Text(productType.actionSheetTitle)
-                        .bodyStyle()
-                    Text(productType.actionSheetDescription)
-                        .subheadlineStyle()
+        ForEach(groupedProductTypes, id: \.0) { (category, productTypes) in
+            VStack {
+                Text(category.label)
+                    .subheadlineStyle()
+                    .textCase(.uppercase)
+                    .frame(maxWidth: .infinity, alignment: .leading)
+                    .padding(.bottom, Constants.categoryVerticalSpacing)
+                    .padding(.horizontal, Constants.horizontalSpacing)
+
+                ForEach(productTypes, id: \.self) { productType in
```

Comment:
Nit: `BottomSheetProductType` is already conforming to `Identifiable`, so we can omit the `id` parameter here.

Reply:
👍🏼  updated in https://github.com/woocommerce/woocommerce-ios/pull/12273/commits/34442d134d84058e62685dcfb68a6ecdfc1bae34